src/actions/getJwtToken.ts

```diff
@@ -2,12 +2,47 @@ import { functions } from '../firebase/firebase'
 import { UID } from '../../types/uid'
 import { JWTToken } from '../../types/JWT'
 
+const RETRY = {
+    number: 4,
+    interval: 300,
+}
+
+async function tryAtMost(promise, maxRetries, retryInterval) {
+    let retriesNumber = 0
+    while (retriesNumber <= maxRetries) {
+        try {
+            const result = await promise
+            return result
+        } catch (err) {
+            retriesNumber++
+            await new Promise((resolve) => {
+                setTimeout(function () {
+                    resolve(true)
+                }, retryInterval)
+            })
+        }
+    }
+
+    throw new Error()
+}
+
 export const getJwtToken = async (uid: UID): Promise<JWTToken> => {
     try {
-        const result = await functions.getToken({
+        const getTokenPromise = functions.getToken({
             uid,
         })
 
+        /*
+            tryAtMost is needed because before a token can exist, firebase functions hook
```

Comment:
`firebase functions hook` > `Firebase user create trigger need to successfully...` is more precise